### PR TITLE
Solving (all but 1) level -2 parser/compiler warnings, compiling with XC8 (v2.32)

### PR DESCRIPTION
**Pull request by PA1RB**
- Removed all "far" qualifiers as per [MPLAB® C18 to XC8 C Compiler Migration Guide](https://ww1.microchip.com/downloads/en/DeviceDoc/50002184B.pdf), section 3.9.1 and 3.10.1
- Altered few data types / typecasts

Code now compiles with just 1 warning  when compiled with XC8 Compiler Warning level -2: 

- _(765) on line 625 of modbus.c_

Code compiled successfully with:
- MPLAB X IDE v5.45
- XC8 (v2.32)
- Pack,PIC18F-K_DFP,1.1.58

...and seems to execute just fine on the target :-)

### CODE REVIEW

File: SmartEVSE2.X/EVSE.c [hunk 1]
```diff
@@ -1425,8 +1425,8 @@ unsigned int getItemValue(unsigned char nav) {
  * @param unsigned char nav
  * @return unsigned char[] MenuItemOption
  */
-const far char * getMenuItemOption(unsigned char nav) {
-    unsigned char Str[10];
+const char * getMenuItemOption(unsigned char nav) {
+    static unsigned char Str[10];
```
Comment (PA1RB) [inline SmartEVSE2.X/EVSE.c:1429]: added static qualifier as this pointer is used as a return value and could be destroyed after return

Comment (maximevince) [reply SmartEVSE2.X/EVSE.c:1429]: Exactly: https://github.com/SmartEVSE/SmartEVSE-2/pull/6/commits/5814af6280e51cbbfc12d920fbc62411f54a0045#diff-3f22bc2aa11b85071fa00c848a58a870628ef4428ebec048596213bd13831780R1439
:)

Comment (PA1RB) [reply SmartEVSE2.X/EVSE.c:1429]: And you have the better comment!

File: SmartEVSE2.X/EVSE.c [hunk 2]
```diff
@@ -242,7 +242,7 @@ unsigned char OldButtonState = 0x0f;
 unsigned char LCDNav = 0;
 unsigned char SubMenu = 0;
 unsigned long ScrollTimer = 0;
-unsigned char LCDpos = 0;
+char LCDpos = 0;
```
Comment (mstegen) [inline SmartEVSE2.X/EVSE.c:245]: Why unsigned char to signed char?

File: SmartEVSE2.X/EVSE.c (hunk 1 above)
Comment (mstegen) [reply SmartEVSE2.X/EVSE.c:1429]: Thanks , both!

File: SmartEVSE2.X/modbus.c [hunk 3]
```diff
@@ -457,9 +457,9 @@ signed long receiveMeasurement(unsigned char *buf, unsigned char pos, unsigned c
     if (IsDouble) {
         combineBytes(&dCombined, buf, pos, Endianness);
         if (Divisor >= 0) {
-            lCombined = dCombined / (signed long)pow10[Divisor];
```
Comment (PA1RB) [inline SmartEVSE2.X/modbus.c:460]: 2x typecast added as compiler throws warning _(356) implicit conversion of float to integer when executed_ at warning level -2

File: SmartEVSE2.X/EVSE.c (hunk 2 above)
Comment (PA1RB) [reply SmartEVSE2.X/EVSE.c:245]: Went back to Master version and removed all "far" qualifiers.
Could not reproduce any issue with LCDpos being unsigned char, so indeed no need to change!
Therefore also no need to change in EVSE.h
Please disregard!

File: SmartEVSE2.X/modbus.c (hunk 3 above)
Comment (mstegen) [reply SmartEVSE2.X/modbus.c:460]: My XC8 compiler (v2.10) doesn't complain about this, but thanks for the fix.

**Comment by mstegen** [comment]
Thanks for your contribution @PA1RB, we can probably merge it now.

**Review by mstegen — APPROVED** [review]
Thanks again for your contribution

**Pull request merged**